pkgs/stdenv/generic/setup.sh

```diff
@@ -1,6 +1,12 @@
 set -eu
 set -o pipefail
 
+if [ -n "${BASH_VERSINFO-}" ] && [ "${BASH_VERSINFO-}" -lt 4 ]; then
```

Comment:
```suggestion
if [[ ${BASH_VERSINFO:0} -lt 4 ]]; then
```

Reply:
In this case, we still need to support zsh which will not have BASH_VERSINFO set. I guess we could do `${BASH_VERSINFO:5}`, but that's kind of weird.